README.md

```diff
@@ -1 +1,293 @@
+ * `description` (string) - Brief description of what the migrations does.
+ * `migrate` (function) - Function that performs the migration (see signature of this function below)
+ * `revert` (function) - If defined then this function will revert the migration to the previous version. Otherwise it is assumed that it is not possible to revert this migration.
+
+##### `migrate(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').migrate()` function that drives the whole migration process!_
+
+Arguments:
+ * `repoPath` (string) - absolute path to the root of the repo
+ * `options` (object, optional) - object containing `IPFSRepo` options, that should be used to construct a datastore instance.
+ * `isBrowser` (bool) - indicates if the migration is run in a browser environment (as opposed to NodeJS)
+ 
+##### `revert(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').revert()` function that drives the whole backward migration process!_
+
+Arguments:
+ * `repoPath` (string) - path to the root of the repo
+ * `options` (object, optional) - object containing `IPFSRepo` options, that should be used to construct the datastore instance.
+ * `isBrowser` (bool) - indicates if the migration is run in a browser environment (as opposed to NodeJS)
+
+#### Browser vs. NodeJS environments
+
+The migration might need to distinguish in which environment it runs (browser vs. NodeJS). For this reason there is an argument
+`isBrowser` passed to migrations functions. But with simple migrations it should not be necessary to distinguish between
+these environments as the datastore implementation will handle the main differences. 
+
+There are currently two main datastore implementations:
+ 1. [`datastore-fs`](https://github.com/ipfs/js-datastore-fs) that is backed by file system and is used mainly in the NodeJS environment
+ 2. [`datastore-level`](https://github.com/ipfs/js-datastore-level) that is backed by LevelDB and is used mainly in the browser environment
+ 
+ Both implementations share the same API and hence are interchangeable. 
+
+ When the migration is run in a browser environment, `datastore-fs` is automatically replaced with `datastore-level` even 
+ when it is directly imported (`require('datastore-fs')` will return `datastore-level` in a browser). 
+ So with simple migrations you shouldn't worry about the difference between `datastore-fs` and `datastore-level` 
+ and by default use the `datastore-fs` package (as the replace mechanism does not work vice versa).
+
+#### Guidelines
+
+The recommended way to write a new migration is to first bootstrap a dummy migration using the CLI:
+
+```sh
+> jsipfs-migrations add
+```
+
+A new folder is created with the bootstrapped migration. You can then simply fill in the required fields and 
+write the rest of the migration! 
+
+#### Migration's dependencies
```

Comment:
This section can be removed. I don't think it introduces any new concept to avid js developers that will be doing something as advanced as a repo migration.